gamd/gamdSimulation.py

```diff
@@ -210,43 +246,43 @@ def createGamdSimulation(self, config, platform_name, device_index):
             properties['CudaPrecision'] = 'mixed'
             properties['DeviceIndex'] = device_index
             gamdSimulation.simulation = openmm_app.Simulation(
-                topology.topology, gamdSimulation.system, 
+                topology.topology, gamdSimulation.system,
                 gamdSimulation.integrator, platform, properties)
             gamdSimulation.device_index = device_index
             gamdSimulation.platform = 'CUDA'
         elif user_platform_name == "opencl":
             platform = openmm.Platform.getPlatformByName('OpenCL')
             properties['DeviceIndex'] = device_index
             gamdSimulation.simulation = openmm_app.Simulation(
-                topology.topology, gamdSimulation.system, 
+                topology.topology, gamdSimulation.system,
                 gamdSimulation.integrator, platform, properties)
             gamdSimulation.device_index = device_index
             gamdSimulation.platform = 'OpenCL'
         else:
             platform = openmm.Platform.getPlatformByName(platform_name)
             gamdSimulation.simulation = openmm_app.Simulation(
-                topology.topology, gamdSimulation.system, 
+                topology.topology, gamdSimulation.system,
                 gamdSimulation.integrator, platform)
             gamdSimulation.platform = platform_name
-        
+
         gamdSimulation.simulation.context.setPositions(positions.positions)
-        if box_vectors is not None:
+        if box_vectors is not None and config.input_files.charmm is None:
```

Comment:
I'm don't I understand why this change was made yet .  The box_vectors variable should either contain the necessary parameters, or it should be None indicating it doesn't need to be set.  Can you reply or add a comment why we also need to check that the charm input_files is None?

Reply:
CHARMM systems can only be built by psf.createSystem() if box vectors are assigned beforehand to the psf object (hence the code snippet at line 156) . Thus, this update of box_vectors here is not needed for CHARMM systems and might create issues setting it twice.